hack/cert_generation.sh

```diff
@@ -0,0 +1,267 @@
+authorityKeyIdentifier  = keyid
+
+# CRL extensions exist solely to point to the CA certificate that has issued
+# the CRL.
+
+[ crl_ext ]
+authorityKeyIdentifier  = keyid
+EOF
+}
+
+function sign_cert() {
+  local component=$1
+
+  openssl ca \
+          -in ${WORKING_DIR}/${component}.csr  \
+          -notext                              \
+          -out ${WORKING_DIR}/${component}.crt \
+          -config ${WORKING_DIR}/signing.conf  \
+          -extensions v3_req                   \
+          -batch                               \
+          -extensions server_ext
+}
+
+function generate_cert_config() {
+  local component=$1
+  local extensions=${2:-}
+
+  if [ "$extensions" != "" ]; then
+    cat <<EOF > "${WORKING_DIR}/${component}.conf"
+[ req ]
+default_bits = 4096
+prompt = no
+encrypt_key = yes
+default_md = sha512
+distinguished_name = dn
+req_extensions = req_ext
+[ dn ]
+CN = ${component}
+OU = OpenShift
+O = Logging
+[ req_ext ]
+subjectAltName = ${extensions}
+EOF
+  else
+    cat <<EOF > "${WORKING_DIR}/${component}.conf"
+[ req ]
+default_bits = 4096
+prompt = no
+encrypt_key = yes
+default_md = sha512
+distinguished_name = dn
+[ dn ]
+CN = ${component}
+OU = OpenShift
+O = Logging
+EOF
+  fi
+}
+
+function generate_request() {
+  local component=$1
+
+  openssl req -new                                        \
+          -out ${WORKING_DIR}/${component}.csr            \
+          -newkey rsa:4096                                \
+          -keyout ${WORKING_DIR}/${component}.key         \
+          -config ${WORKING_DIR}/${component}.conf        \
+          -days 712                                       \
+          -nodes
+}
+
+function generate_certs() {
+  local component=$1
+  local extensions=${2:-}
+
+  if [ $REGENERATE_NEEDED = 1 ] || [ ! -f ${WORKING_DIR}/${component}.crt ] || ! openssl x509 -checkend 0 -noout -in ${WORKING_DIR}/${component}.crt; then
+    generate_cert_config $component $extensions
+    generate_request $component
+    sign_cert $component
+  fi
+}
+
+function generate_extensions() {
+  local add_oid=$1
+  local add_localhost=$2
+  shift
+  shift
+  local cert_names=$@
+
+  extension_names=""
+  extension_index=1
+  local use_comma=0
+
+  if [ "$add_localhost" == "true" ]; then
+    extension_names="IP.1:127.0.0.1,DNS.1:localhost"
+    extension_index=2
+    use_comma=1
+  fi
+
+  for name in ${cert_names//,/}; do
+    if [ $use_comma = 1 ]; then
+      extension_names="${extension_names},DNS.${extension_index}:${name}"
+    else
+      extension_names="DNS.${extension_index}:${name}"
+      use_comma=1
+    fi
+    extension_index=$(( extension_index + 1 ))
+  done
+
+  if [ "$add_oid" == "true" ]; then
+    extension_names="${extension_names},RID.1:1.2.3.4.5.5"
+  fi
+
+  echo "$extension_names"
+}
+
+if [ ! -d $WORKING_DIR ]; then
+  mkdir -p $WORKING_DIR
+fi
+
+generate_signing_ca
+init_cert_files
+create_signing_conf
+
+generate_certs 'system.logging.fluentd'
```

Comment:
likely don't need to create the fluend certs here

Reply:
> likely don't need to create the fluend certs here
In the interest of being able to update it when it changes in CLO do we want to remove it?

Reply:
> In the interest of being able to update it when it changes in CLO do we want to remove it?

Not following your question... since we don't test fluentd at all in EO, we dont need to generate its certs here

Reply:
removing